### PR TITLE
Add `ACCOUNT_EMAIL_CHANGED` webhook event

### PR DESCRIPTION
**Pull request by Air-t**
Adds `ACCOUNT_EMAIL_CHANGED` event sent after confirmEmailChange mutation.

Resolves https://github.com/saleor/saleor/issues/13474

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migrations are either absent or optimized for zero downtime
* [ ] The changes are covered by test cases

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -92,9 +92,10 @@ Shipping methods can be removed by the user after it has been assigned to a chec
   - Called when metadata is changed for the Shop object via the generic metadata API or the `shopSettingsUpdate` mutation.
 - Add `CHANNEL_METADATA_UPDATED` webhook - #13448, by @Air-t
   - Called when metadata is changed for the Channel object via the generic metadata API or the `channelUpdate` mutation.
-
 - Add `ACCOUNT_CONFIRMED` webhook - #13471, by @Air-t
   - Called when user confirm an account with `confirmAccount` mutation.
+- Add `ACCOUNT_EMAIL_CHANGED` webhook - #13537, by @Air-t
+  - Called when `aonfirmEmailChange` mutation is triggered.
```
Comment (IKarbowiak) [inline CHANGELOG.md:98]: Typo in `aonfirmEmailChange`

File: saleor/plugins/webhook/plugin.py [hunk 2]
```diff
@@ -239,6 +239,18 @@ def account_change_email_requested(
             new_email,
         )
 
+    def account_email_changed(
+        self,
+        user: "User",
+        previous_value: None,
```
Comment (IKarbowiak) [inline saleor/plugins/webhook/plugin.py:245]: Wrong typing for `previous_value`, it should be rather `Optional["User"]` I think

Comment (Air-t) [reply saleor/plugins/webhook/plugin.py:245]: I guess in this case all of the previous_values types needs a refactor here - the almost only types expected are `None` and `Any` but I see previous_value should represent instance or instences type

**Review by fowczarek — APPROVED** [review]

File: saleor/graphql/webhook/subscription_types.py [hunk 3]
```diff
@@ -214,6 +214,18 @@ def resolve_new_email(root, _info: ResolveInfo):
         return data["new_email"]
 
 
+class AccountEmailChanged(SubscriptionObjectType, AccountOperationBase):
+    new_email = graphene.String(
+        description="The new email address the user wants to change to.",
```
Comment (maarcingebala) [inline saleor/graphql/webhook/subscription_types.py:219]: To be precise, this is the new email address that the user has already changed to, at the point when the event is fired.

Comment (Air-t) [reply saleor/graphql/webhook/subscription_types.py:219]: ah right - I thought it is before the change, but change is already made but this is just a confirmation - so @ that step email is changed already but not confirmed yet

**Review by IKarbowiak — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]

**Review by yellowee — APPROVED** [review]

**Pull request merged**